src/ansys/templates/utils.py

```diff
@@ -123,14 +123,18 @@ def keep_files(files_list, project_path=Path(os.getcwd())):
     # Collect all the files in the project
     all_project_files = project_path.glob("**/*")
 
+    # Fix path name according to OS flavor
+    separator, new_separator = ("/", "\\") if os.name != "posix" else ("/", "/")
+    desired_files = [file.replace(separator, new_separator) for file in files_list]
+
     # Remove all undesired files
     for file in all_project_files:
-        if str(file.relative_to(project_path)) not in files_list and file.is_file():
+        if str(file.relative_to(project_path)).replace(separator, new_separator) not in desired_files and file.is_file():
             file.unlink()
 
     # Remove all undesired directories
     for file in all_project_files:
-        if str(file.relative_to(project_path)) not in files_list and file.is_dir():
+        if str(file.relative_to(project_path)).replace(separator, new_separator) not in desired_files and file.is_dir():
```

Comment:
This second loop is actually useless since the code will never enter inside it. In fact, all_project_files are a generator (not a list or array).  Therefore by iterating through it the first time (L131) it is never rewound. There are multiple approaches to achieving this. One is to put this logic under the first loop such as 

```   
    for file in all_project_files:
        if str(file.relative_to(project_path)).replace(separator, new_separator) not in desired_files:
            if file.is_file():
                file.unlink()
            elif file.is_dir():
                file.rmdir() 
```

Since you are checking whether a directory is the part of the DESIRED_FILES, you should add the folder paths in the input array otherwise, it will try to delete them and some of them would fail since they will not be empty. 

I believe that the right approach is the following:
1. Delete undesired files and collect the folder
2. Parse the folder and delete the empty ones

The code would look like this : 

```
    folders = []
    # Removing undesired files
    for file in all_project_files:
           if str(file.relative_to(project_path)).replace(separator, new_separator) not in desired_files:
            if file.is_file():
                file.unlink()
            elif file.is_dir():
                folders.append(file)

    # Removing undesired folder = empty folder
    for folder in folders:
        if not os.listdir(str(folder)):
            folder.rmdir()
```

Reply:
Thanks for pointing this, @asbfall. You are right, because `all_project_files` is a generator, it gets consumed at L131 thus, not entering next line.

I agree with the code you proposed about collecting folders. The reason why the file and folder loops can't be shared is because it is not possible to remove non-empty directories.

Applying suggested changes 👍🏽